feat: terragrunt support (#7653)
renovatebot/renovate: lib/manager/terragrunt/readme.md

Currently by default, Terragrunt support is limited to Terraform registry sources and github sources that include semver refs, e.g. like `github.com/hashicorp/example?ref=v1.0.0`.

> It is however possible to also use custom [versioning config](../../../docs/usage/configuration-options.md#versioning) in order to support non-semver references. For example, if your want to ref a tag like `module-v1.2.5`, a block like the following would solve this issue:

Reviewer: ```suggestion
It is however possible to also use custom [versioning config](../../../docs/usage/configuration-options.md#versioning) in order to support non-semver references.
For example, if your want to ref a tag like `module-v1.2.5`, a block like the following would solve this issue:
```

This suggestion makes it so we conform to the one sentence per line style.

> Currently by default, Terragrunt support is limited to Terraform registry sources and github sources that include semver refs, e.g. like `github.com/hashicorp/example?ref=v1.0.0`.

Reviewer: ```suggestion
Currently by default, Terragrunt support is limited to Terraform registry sources and GitHub sources that include semver refs, e.g. like `github.com/hashicorp/example?ref=v1.0.0`.
```

GitHub is a noun and should be capitalized.

> It is however possible to also use custom [versioning config](../../../docs/usage/configuration-options.md#versioning) in order to support non-semver references.

Author: Yep, shorter and concise

> For example, if your want to ref a tag like `module-v1.2.5`, a block like the following would solve this issue:

Reviewer: ```suggestion
For example, if you want to reference a tag like `module-v1.2.5`, a block like this would work:
```

Rewrite sentence for clarity.

```json
"terraform": {
	"versioning": "regex:^((?<compatibility>.*)-v|v*)(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)$"
}
```

> Fixed Terragrunt versions like the following will receive a PR whenever there is a newer version available:

Reviewer: It's not clear to me what this "Fixed Terragrunt versions" is talking about.
The word fixed brings to mind things like: "Oh it was broken, but now it works." or "I'm hard-coding a version here".
I don't know how to improve this sentence.

Maybe somebody else has a good suggestion for this sentence?

Author: @HonkingGoose the same is also in `terraform` manager. But anyway, to make it more clear I suggest this: `Pinned Terragrunt dependecies like the following will receive a PR whenever there is a newer version available:`

Reviewer: Hmm, I'm not sure if the word "pinned" is better. The word pinned is usually used to mean: "Use one this specific version", at least the way dependency management bots use the word like this.

Author: And that is the exact case here: you need to pin to a version (e.g use this specific version) in order for `renovate` to work and suggest new version available

Reviewer: > And that is the exact case here: you need to pin to a version (e.g use this specific version) in order for renovate to work and suggest new version available

---

```suggestion
Pinned Terragrunt dependencies like the the following will receive a PR whenever there is a newer version available:
```

Let's go with this then. :+1:
I fixed a small spelling error in your comment's suggestion with the word `dependencies`.

```hcl
terraform {

>   source = "github.com/hashicorp/example?ref=v1.0.0"

Reviewer: At line 1 in this document you say: "GitHub sources that include semver refs, e.g. like `github.com/hashicorp/example?ref=v1.0.0`."

But why do we use `source = "github.com/hashicorp/example?ref=v1.0.0"` in the "Fixed Terragrunt versions" example then? Are you sure that makes sense?

It's a bit confusing to me, to first say "Oh this works by default", and then later in a an code block say: "Oh wait you need to fix this by doing this thing".

Author: Sorry but I don't understand what the problem is. It's same GitHub source like in the example.

Author: Furthermore, people using terragrunt or terraform need to know how to use these tools before using renovate.

Reviewer: Let's "resolve" this then, you're the maker of the code, so you know best if the docs are correct. :wink: 

Can you click the "resolve" button on this conversation? I don't have the rights to mark this as "resolved" myself. :smile:

Author: Sure thing :)

> Currently Terragrunt support is limited to Terraform registry sources and github sources that include semver refs, e.g. like `github.com/hashicorp/example?ref=v1.0.0`.

Reviewer 2: I'm interested to know how many other types there are which we might want to support in future?

Author: I've updated documentation to include info about custom config for using ref which are not in line with semver.

The text you mentioned is basically the same as for Terraform, but just FYI all the available source types supported by both of them can be [found here](https://www.terraform.io/docs/modules/sources.html)
